### PR TITLE
fix: Pass explicit length in Struct makeData calls

### LINKED ISSUE #403: makeData for Struct with empty children silently defaults length to 0
## Description

In `src/data.ts`, `visitStruct` computes a default `length` via:

```typescript
length = children.reduce((len, { length }) => Math.max(len, length), 0),
```

When `children` is `[]` and `props.length` is omitted, `reduce` returns `0`. This silently discards any intended row count for zero-child Struct data.

## Impact

Two callers in `src/recordbatch.ts` omit `length` when calling `makeData` for Struct types:

- **`setChildAt`** (line 264): `makeData({ type: new Struct<T>(fields), children })` — should pass `length: data.length` to preserve the batch's row count
- **`_InternalEmptyPlaceholderRecordBatch`** (line 383): `makeData({ type: new Struct<T>(schema.fields), nullCount: 0, children })` — should pass `length: 0` explicitly

Both work today because they always have children to reduce over, but they're fragile — any future change that results in empty children would silently drop the row count.

## Proposed Fix

Pass explicit `length` in both `makeData` call sites so they don't rely on the reduce-over-children fallback.

### PR DESCRIPTION
**Pull request by rustyconover**
## Summary
- Pass `length: data.length` in `setChildAt` to preserve the batch's row count when constructing Struct data
- Pass `length: 0` explicitly in `_InternalEmptyPlaceholderRecordBatch` instead of relying on reduce-over-children fallback
- Add tests for Struct `makeData` with empty children and `setChildAt` row count preservation

Closes #403

## Test plan
- [x] `makeData` with empty children preserves explicit length
- [x] `makeData` with empty children and no length defaults to 0
- [x] `setChildAt` preserves `numRows` after column replacement
- [x] Full test suite passes (`yarn test -t src`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/unit/data/struct-zero-children-tests.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    makeData, makeVector,
+    Schema, Field, RecordBatch,
+    Struct, Int32
+} from 'apache-arrow';
+
+describe('Struct with zero children', () => {
+
+    test('makeData with empty children preserves explicit length', () => {
+        const data = makeData({ type: new Struct([]), length: 50, nullCount: 0, children: [] });
+        expect(data).toHaveLength(50);
```
Comment (trxcllnt) [inline test/unit/data/struct-zero-children-tests.ts:30]: I have the same question here as https://github.com/apache/arrow-js/pull/402#pullrequestreview-3893887624.

Comment (rustyconover) [reply test/unit/data/struct-zero-children-tests.ts:30]: See the reply there, this bug was discovered in the same use case.
